docs/references/metadata.md

```diff
@@ -0,0 +1,274 @@
+with Alteryx Designer. The metadata is used to map this data during the
+conversion to or from the internal Designer data format ([Alteryx
+Multi-threaded
+Processing](https://help.alteryx.com/20223/designer/alteryx-amp-engine) or AMP).
+
+## Metadata Description
+
+-   **name**: This is the same as the column name in a dataset.
+-   **type**: The internal AMP Designer type.
+-   **size**: The size of the data, in bytes.
+-   **scale**: Scale is used only for 1 type:
+    `fixeddecimal`.
+-   **source**: This is a string that describes the origin of the data
+    (for example, if data is read from a file in Designer, this is set
+    to the file path of this input file).
+-   **description**: This is an all-purpose string field.
+
+## Why Use Metadata?
+
+It\'s not mandatory to use metadata in Python plugins. The simplest
+method is typically to create a Python Arrow schema directly, as the
+engine derives the necessary types when sent to Designer. But some
+plugins might need to export output data types to Designer at plugin
+creation to build the workflow. In this case, metadata has to be sent to
+Designer through the `__init__` method.
+
+Also, sometimes you might need to fine-tune the data type used by
+Designer. Metadata is the only way to be sure to map a precise type of
+data to Designer because Python Arrow types might not allow for that.
+You might also want to modify fields like source or description to send
+some specific information through the Designer pipeline. Input metadata
+can also be used by the plugin for any reason, for example, the source
+field could be a file path the plugin can use.
+
+## Usage
+
+You can use metadata in different places.
+
+### \_\_init\_\_ Method
+
+This method is called at plugin creation when Designer requests the
+output data types from the plugin before the workflow runs. You can send
+metadata to Designer at this time.
+
+#### Example
+
+In this example, the data schema is a table with 2 columns of types
+`int16` and `string`.
+
+    def __init__(self, provider: AMPProviderV2):
+        provider.push_outgoing_metadata("Output", create_schema({
+            "volts": {
+                "type": FieldType.int16
+            },
+            "device" {
+                "type": FieldType.string
+            }
+        }))
+
+### on_record_batch Method
+
+This method is called each time Designer sends data through the plugin.
+You can extract metadata from the input data for information and modify
+output data metadata. However, we don\'t recommend modifying the type,
+size, and scale as unexpected results can occur in Designer. You can
+modify the source and description.
+
+### on_complete Method
+
+On input plugins, a complete schema created with metadata can be
+exported to Designer for a precise data schema.
+
+## Use Cases
+
+### Define a Schema at \_\_init\_\_
+
+In this example, we specify type `int16` for
+column `volts` and type
+`string` for column
+`device`.
```

Comment:
TW review:

This example specifies type `int16` for
column `volts` and type
`string` for column
`device`.

Reply:
fixed